src/gtest/test_sideChain.cpp

```diff
@@ -0,0 +1,1196 @@
+        <<"Following flush, persisted fwd amount should equal the one in view";
+}
+
+TEST_F(SideChainTestSuite, EmptyFlushDoesNotPersistNewSideChain) {
+    const Sidechain::ScInfoMap & initialScCollection = sideChainManager.getScInfoMap();
+
+    //prerequisites
+    ASSERT_TRUE(coinViewCache.getScInfoMap().size() == 0)<<"There should be no new txs to persist";
+    ASSERT_TRUE(initialScCollection.size() == 0)<<"Test requires no sidechains initially";
+
+    //test
+    bool res = coinViewCache.Flush();
+
+    //checks
+    EXPECT_TRUE(res);
+
+    const Sidechain::ScInfoMap & finalScCollection = sideChainManager.getScInfoMap();
+    EXPECT_TRUE(finalScCollection == initialScCollection)
+        <<"Sidechains collection should not have changed with empty flush";
+}
+
+TEST_F(SideChainTestSuite, EmptyFlushDoesNotAlterExistingSideChainsCollection) {
+    //prerequisites
+    preFillSidechainsCollection();
+
+    const Sidechain::ScInfoMap & initialScCollection = sideChainManager.getScInfoMap();
+
+    ASSERT_TRUE(coinViewCache.getScInfoMap().size() == 0)<<"There should be no new txs to persist";
+    ASSERT_TRUE(initialScCollection.size() != 0)<<"Test requires some sidechains initially";
+
+    //test
+    bool res = coinViewCache.Flush();
+
+    //checks
+    EXPECT_TRUE(res);
+
+    const Sidechain::ScInfoMap & finalScCollection = sideChainManager.getScInfoMap();
+    EXPECT_TRUE(finalScCollection == initialScCollection)
+        <<"Sidechains collection should not have changed with empty flush";
+}
+
+///////////////////////////////////////////////////////////////////////////////
+/////////////////////////////// Structural UTs ////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////
+TEST_F(SideChainTestSuite, ManagerIsSingleton) {
+    //test
+    Sidechain::ScMgr& rAnotherScMgrInstance = Sidechain::ScMgr::instance();
+
+    //check
+    EXPECT_TRUE(&sideChainManager == &rAnotherScMgrInstance)
+            << "ScManager Instances have different address:"
+            << &sideChainManager << " and " << &rAnotherScMgrInstance;
+}
+
+TEST_F(SideChainTestSuite, ManagerDoubleInitializationIsForbidden) {
+    size_t cacheSize(0);
+    bool fWipe(false);
+
+    //prerequisites: first initialization happens in fixture's setup
+
+    //test
+    bool bRet = sideChainManager.initialUpdateFromDb(cacheSize, fWipe, Sidechain::ScMgr::mock);
+
+    //Checks
+    EXPECT_FALSE(bRet) << "Db double initialization should be forbidden";
+}
+
+///////////////////////////////////////////////////////////////////////////////
+////////////////////////// Test Fixture definitions ///////////////////////////
+///////////////////////////////////////////////////////////////////////////////
+void SideChainTestSuite::resetBaseParams() {
+    //force reset of pCurrentBaseParams
+    CBaseChainParams* nakedCurrentBaseParams = &const_cast<CBaseChainParams &>(BaseParams());
```

Comment:
What is the need of nakedCurrentBaseParams (and nakedCurrentParams below)?

Reply:
It's a brute force way to get access to a static pointers pCurrentBaseParams/pCurrentParams defined in chainparamsbase.cpp/chainparams.cpp, whose visibility is limited to those files.

My purpose was initially to make sure I reset those parameters to null following my tests. However:
1. I do not know if the state of those parameters is really null. This is because these tests lives in the same binary as others, which may have set it already.
2. The SelectBaseParams()/SelectParams() functions already take care of setting the parameters to the right one at the beginning of the test, regardless the state of parameters before it

I have removed the reset functions and verified they do not affect test correctness